bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosBridgeHandler.java

```diff
@@ -0,0 +1,519 @@
+            logger.debug("Not disposing bridge because of missing apiConnection");
+            return;
+        }
+
+        localApiConnection.unregisterForChangeEvents(this);
+        logger.debug("HEOS bridge removed from change notifications");
+
+        logger.debug("Dispose bridge '{}'", thing.getProperties().get(PROP_NAME));
+        localApiConnection.closeConnection();
+    }
+
+    /**
+     * Manages the removal of the player or group channels from the bridge.
+     */
+    @Override
+    public synchronized void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        logger.debug("Disposing child handler for: {}.", childThing.getUID().getId());
+        if (bridgeHandlerDisposalOngoing) { // Checks if bridgeHandler is going to disposed (by stopping the binding or
+            // openHAB for example) and prevents it from being updated which stops the
+            // disposal process.
+        } else if (childHandler instanceof HeosPlayerHandler) {
+            String channelIdentifier = "P" + childThing.getUID().getId();
+            updateThingChannels(channelManager.removeSingleChannel(channelIdentifier));
+        } else if (childHandler instanceof HeosGroupHandler) {
+            String channelIdentifier = "G" + childThing.getUID().getId();
+            updateThingChannels(channelManager.removeSingleChannel(channelIdentifier));
+            // removes the handler from the groupMemberMap that handler is no longer called
+            // if group is getting online
+            removeGroupHandlerInformation((HeosGroupHandler) childHandler);
+        }
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        logger.debug("Initialized child handler for: {}.", childThing.getUID().getId());
+        childHandlerInitializedFutures.add(scheduler.submit(() -> addPlayerChannel(childThing, null)));
+    }
+
+    void resetPlayerList(ChannelUID channelUID) {
+        selectedPlayerList.forEach(element -> updateState(element[1], OnOffType.OFF));
+        selectedPlayerList.clear();
+        updateState(channelUID, OnOffType.OFF);
+    }
+
+    /**
+     * Sets the HEOS Thing offline
+     */
+    @SuppressWarnings("null")
+    public void setGroupOffline(String hashValue) {
+        HeosGroupHandler groupHandler = groupHandlerMap.get(hashValue);
+        if (groupHandler != null) {
+            groupHandler.setStatusOffline();
+        }
+    }
+
+    /**
+     * Sets the HEOS Thing online. Also updates the link between
+     * the groupMemberHash value with the actual gid of this group *
+     */
+    public void setGroupOnline(String groupMemberHash, String groupId) {
+        hashToGidMap.put(groupMemberHash, groupId);
+        groupHandlerMap.forEach((hash, handler) -> {
+            if (hash.equals(groupMemberHash)) {
+                handler.setStatusOnline();
+                addPlayerChannel(handler.getThing(), groupId);
+            }
+        });
```

Comment:
You could do this without using an Optional but using an Optional here is the easiest way to make the null checker happy.
```suggestion
        Optional.ofNullable(groupHandlerMap.get(groupMemberHash))
            .ifPresent(handler -> {
                handler.setStatusOnline();
                addPlayerChannel(handler.getThing(), groupId);
        });
```